records/state/state.go

```diff
@@ -162,7 +162,8 @@ func statusIPs(st []Status, src func(*Status) []string) []string {
 	// https://github.com/apache/mesos/blob/0.24.0/src/slave/slave.cpp#L5226-L5238
 	ts, j := -1.0, -1
 	for i := range st {
-		if st[i].State == "TASK_RUNNING" && st[i].Timestamp > ts {
+		state := st[i].State
+		if (state == "TASK_RUNNING" || state == "TASK_STAGING") && st[i].Timestamp > ts {
```

Comment:
"This doesn't mean that it can guarantee healthiness of the returned service instances, only that it does its best to direct clients to capable ones."

I think an appropriate interpretation of doing its best to direct clients to capable nodes would be the following:

Give the DNS address to the most recently updated RUNNING Task.  If there are not RUNNING Tasks, give it to the most recently update STAGING Task.  Otherwise, generate no DNS address.

I think it's a good point that a STAGING Task probably will not be able to respond to clients, and if a better alternative is available in a RUNNING Task, it should be provided.